src/query/processor/cypher/astbuilder/ASTBuilder.cpp

```diff
@@ -0,0 +1,1701 @@
+
+any visitOC_NonArithmeticOperatorExpression(CypherParser::OC_NonArithmeticOperatorExpressionContext *ctx)  ;
+
+any visitOC_ListOperatorExpression(CypherParser::OC_ListOperatorExpressionContext *ctx)  ;
+
+any visitOC_PropertyLookup(CypherParser::OC_PropertyLookupContext *ctx)  ;
+
+any visitOC_Atom(CypherParser::OC_AtomContext *ctx)  ;
+
+any visitOC_CaseExpression(CypherParser::OC_CaseExpressionContext *ctx)  ;
+
+any visitOC_CaseAlternative(CypherParser::OC_CaseAlternativeContext *ctx)  ;
+
+any visitOC_ListComprehension(CypherParser::OC_ListComprehensionContext *ctx)  ;
+
+any visitOC_PatternComprehension(CypherParser::OC_PatternComprehensionContext *ctx)  ;
+
+any visitOC_Quantifier(CypherParser::OC_QuantifierContext *ctx)  ;
+
+any visitOC_FilterExpression(CypherParser::OC_FilterExpressionContext *ctx)  ;
+
+any visitOC_PatternPredicate(CypherParser::OC_PatternPredicateContext *ctx)  ;
+
+any visitOC_ParenthesizedExpression(CypherParser::OC_ParenthesizedExpressionContext *ctx)  ;
+
+any visitOC_IdInColl(CypherParser::OC_IdInCollContext *ctx)  ;
+
+any visitOC_FunctionInvocation(CypherParser::OC_FunctionInvocationContext *ctx)  ;
+
+any visitOC_FunctionName(CypherParser::OC_FunctionNameContext *ctx)  ;
+
+any visitOC_ExistentialSubquery(CypherParser::OC_ExistentialSubqueryContext *ctx)  ;
+
+any visitOC_ExplicitProcedureInvocation(CypherParser::OC_ExplicitProcedureInvocationContext *ctx)  ;
+
+any visitOC_ImplicitProcedureInvocation(CypherParser::OC_ImplicitProcedureInvocationContext *ctx)  ;
+
+any visitOC_ProcedureResultField(CypherParser::OC_ProcedureResultFieldContext *ctx)  ;
+
+any visitOC_ProcedureName(CypherParser::OC_ProcedureNameContext *ctx)  ;
+
+any visitOC_Namespace(CypherParser::OC_NamespaceContext *ctx)  ;
+
+any visitOC_Variable(CypherParser::OC_VariableContext *ctx)  ;
+
+any visitOC_Literal(CypherParser::OC_LiteralContext *ctx)  ;
+
+any visitOC_BooleanLiteral(CypherParser::OC_BooleanLiteralContext *ctx)  ;
+
+any visitOC_NumberLiteral(CypherParser::OC_NumberLiteralContext *ctx)  ;
+
+any visitOC_IntegerLiteral(CypherParser::OC_IntegerLiteralContext *ctx)  ;
+
+any visitOC_DoubleLiteral(CypherParser::OC_DoubleLiteralContext *ctx)  ;
+
+any visitOC_ListLiteral(CypherParser::OC_ListLiteralContext *ctx)  ;
+
+any visitOC_MapLiteral(CypherParser::OC_MapLiteralContext *ctx)  ;
+
+any visitOC_PropertyKeyName(CypherParser::OC_PropertyKeyNameContext *ctx)  ;
+
+any visitOC_Parameter(CypherParser::OC_ParameterContext *ctx)  ;
+
+any visitOC_SchemaName(CypherParser::OC_SchemaNameContext *ctx)  ;
+
+any visitOC_ReservedWord(CypherParser::OC_ReservedWordContext *ctx)  ;
+
+any visitOC_SymbolicName(CypherParser::OC_SymbolicNameContext *ctx)  ;
+
+any visitOC_LeftArrowHead(CypherParser::OC_LeftArrowHeadContext *ctx)  ;
+
+any visitOC_RightArrowHead(CypherParser::OC_RightArrowHeadContext *ctx)  ;
+
+any visitOC_Dash(CypherParser::OC_DashContext *ctx)  ;
+
+
+
+any ASTBuilder::visitOC_Cypher(CypherParser::OC_CypherContext *ctx)  {
```

Comment:
Lets arrange these function declarations also based on the order of the function definitions appearance.

Reply:
the order is very similar to hierarchy of the cypher grammar rule set. and this order was defined by antlr4 then it can be helpful for debugging. if it is necessary, I can re-arrange it.

Reply:
In this case lets not re-order the functions.

Reply:
okay sir.